pkg/cloud/services/eks/nodegroup.go

```diff
@@ -337,17 +338,38 @@ func (s *NodegroupService) reconcileNodegroupVersion(ng *eks.Nodegroup) error {
 			updateMsg = fmt.Sprintf("to AMI version %s", *input.ReleaseVersion)
 		}
 
+		conditions.MarkFalse(
+			s.scope.ManagedMachinePool,
+			expinfrav1.EKSNodegroupUpdateSucceededCondition,
+			expinfrav1.EKSNodegroupUpdatingReason,
+			clusterv1.ConditionSeverityInfo,
+			"",
+		)
 		if err := wait.WaitForWithRetryable(wait.NewBackoff(), func() (bool, error) {
 			if _, err := s.EKSClient.UpdateNodegroupVersion(input); err != nil {
+				conditions.MarkFalse(
+					s.scope.ManagedMachinePool,
+					expinfrav1.EKSNodegroupUpdateSucceededCondition,
+					expinfrav1.EKSNodegroupFailedToUpdateReason,
+					clusterv1.ConditionSeverityError,
+					"",
+				)
 				if aerr, ok := err.(awserr.Error); ok {
 					return false, aerr
 				}
 				return false, err
 			}
-			record.Eventf(s.scope.ManagedMachinePool, "SuccessfulUpdateEKSNodegroup", "Updated EKS nodegroup %s %s", eksClusterName, updateMsg)
+			record.Eventf(s.scope.ManagedMachinePool, "UpdatingEKSNodegroup", "Updating EKS nodegroup %s %s", eksClusterName, updateMsg)
 			return true, nil
 		}); err != nil {
 			record.Warnf(s.scope.ManagedMachinePool, "FailedUpdateEKSNodegroup", "failed to update the EKS nodegroup %s %s: %v", eksClusterName, updateMsg, err)
+			conditions.MarkFalse(
+				s.scope.ManagedMachinePool,
+				expinfrav1.EKSNodegroupUpdateSucceededCondition,
+				expinfrav1.EKSNodegroupFailedToUpdateReason,
+				clusterv1.ConditionSeverityError,
+				"",
```

Comment:
We should probably set the error message here so that we can see this when we do a describe.

Reply:
Updated